app/src/main/java/agersant/polaris/api/remote/PolarisExoPlayerDataSourceFactory.java

```diff
@@ -66,7 +66,7 @@ public void onTransferEnd(DataSource source, DataSpec dataSpec, boolean isNetwor
         private RandomAccessFile file;
 
         PolarisExoPlayerHttpDataSource(OfflineCache offlineCache, RequestProperties requestProperties, PolarisExoPlayerTransferListener listener, File scratchLocation, CollectionItem item) {
-            super("Polaris Android", null, DEFAULT_CONNECT_TIMEOUT_MILLIS, DEFAULT_READ_TIMEOUT_MILLIS, true, requestProperties);
+            super("Polaris Android", DEFAULT_CONNECT_TIMEOUT_MILLIS, DEFAULT_READ_TIMEOUT_MILLIS, true, requestProperties); //TODO: use factory instead of deprecated constructor
```

Comment:
Assuming it isn't a very large change, could you do this upgrade so we don't keep deprecated code around?

Reading ExoPlayer release notes, I see they now have a playlist API which might be good to investigate. I think Polaris could benefit from it in terms of achieving gapless playback and potentially having less Polaris code. I'll write a new issue for that one, no need to worry about it in this PR.

Reply:
I have rough plans on upgrading to `androidx.media2` which if I remember correctly `Exoplayer` has an extension for, that integrates quite nicely. The playlist API sounds interesting and I will definitely take a look at that.
The problem is that the deprecated constructor is being replaced by a Factory. Since were extending `DefaultHttpDataSource` that seemed like too much trouble right now.